pyquil/quil.py

```diff
@@ -341,7 +342,7 @@ def measure_all(self, *qubit_reg_pairs):
                 self.inst(MEASURE(qubit_index, classical_reg))
         return self
 
-    def while_do(self, classical_reg, q_program):
+    def while_do(self, classical_reg: int, q_program: 'Program') -> 'Program':
```

Comment:
I think the documented type for `classical_reg` here was wrong and it should be `MemoryReference`, instead. It looks like the `JumpUnless` constructor will raise a `TypeError` if `classical_reg` isn't a `MemoryReference`. 

Per my comment on `if_then`, below, we should update this to call `unpack_classical_reg` and update the type hint for `classical_reg` to be `MemoryReferenceDesignator`.